#Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении
Выполнили студенты гр. 0381 Лаари Екатерина и Захаров Филипп. Вариант №9

## Цель работы

Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения

Интервальной называется статистическая оценка, которая задается двумя числами - концами интервала (в котором может находиться значение параметра $θ$

Точностью $θ^*$ - статистической оценки параметра $θ$ называется значение $δ$, удовлетворяющее неравенству: $$δ > |θ^* - θ|$$

Надежностью оценки $θ^*$ называется вероятность $γ$, с которой выполняется неравенство: $$γ = P\{δ > |θ^* - θ|\} $$

Интервал $$(θ^* - δ ; θ^* + δ)$$ называется доверительным интервалом и является интервальной оценкой параметра $θ$

Значение $α = 1 - γ$ называют уровнем значимости. Значение γ задают обычно равным 0.95, 0.99, 0.995, 0.999

Статистической гипотезой называют гипотезу о виде неизвестного распределения исследуемой случайной величины или о значении параметров известных распределений.

Выдвинутую гипотезу называют нулевой или основной гипотезой и обозначают $H_0$

Конкурирующей или альтернативной называют гипотезу $H_1$, которая противоречит основной гипотезе.

Ошибка первого рода состоит в том, что нулевая гипотеза отвергается, хотя она верна

Ошибка второго рода состоит в том, что нулевая гипотеза принимается, хотя она неверна

Критерий $χ^2$. Согласно этому критерию на основании выборочных данных вычисляется "наблюденное" значение случайной величины $χ^2$:
$$χ^2_{набл} = \sum^K_{i=1} \frac{(n_i - n_i')^2}{n_i'} (5)$$
Здесь $n_i'$ - теоретические частоты - частоты, с которыми нормально распределенная величина с параметрами $\overline{x_в}$ и $S$ попала бы в $i$-й интервал интервального ряда при проведении $N$ испытаний

## Постановка задачи

Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона $\chi^2$. Дать содержательную интерпретацию полученным результатам.

## Выполнение работы

1. Вычислить точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности γ∈{0.95,0.99}. Сделать выводы.

$γ∈\{0.95, 0.99\}$

Из прошлой практической работы:\
$\overline X = 449.79$\
$S = 57.11$\
$N=100$

Используя таблицу значений коэффициентов Стьюдента, для

$\gamma = 0.95\\ N = 99 \\ t_\gamma=1.984$

Доверительный интервал для математического ожидания

для

$\gamma = 0.99\\ N = 99 \\ t_\gamma=2.626$

Доверительный интервал для математического ожидания

$(\overline X - t_\gamma\frac{S}{\sqrt{N}}, \overline X + t_\gamma\frac{S}{\sqrt{N}})$


In [ ]:
X = 449.79
S = 57.11
N = 100
print("Для gamma == 0.95")
print("(", round(X - 1.984 * S / N ** 0.5, 2), ", ", round(X + 1.984 * S / N ** 0.5, 2), ")", sep='')

print("Для gamma == 0.99")
print("(", round(X - 2.626 * S / N ** 0.5, 2), ", ", round(X + 2.626 * S / N ** 0.5, 2), ")", sep='')

Для gamma == 0.95
(438.46, 461.12)
Для gamma == 0.99
(434.79, 464.79)


Истинное значение математического ожидания содержится в интервале  (438.46, 461.12)  с вероятностью  0.95  и в интервале  (434.79, 464.79)  с вероятностью  0.99 .

2. Для вычисления границ доверительного интервала для среднеквадратичного отклонения определить значение q при заданных γ и n. Построить доверительные интервалы, сделать выводы.

По таблице значений q для $\gamma = 0.95, \gamma = 0.99, N=100$

$q = 0.143\ q = 0.198$

$σ∈(S(1-q);  S(1+q))$

In [ ]:
print("Для q == 0.143")
q = 0.143
print("(", round(S * (1 - q), 2), ", ", round(S * (1 + q), 2), ")", sep='')

print("Для q == 0.198")
q = 0.198
print("(", round(S * (1 - q), 2), ", ", round(S * (1 + q), 2), ")", sep='')

Для q == 0.143
(48.94, 65.28)
Для q == 0.198
(45.8, 68.42)


То есть, согласно вычислениям, при надежности $0.95$ значение $q$ равно $0.143$, а доверительный интервал $(48.94, 65.28)$

А при при надежности $0.99$ значение $q$ равно $0.198$, а доверительный интервал $(45.8, 68.42)$

3. Проверить гипотезу о нормальности заданного распределения с помощью критерия χ2 (Пирсона). Для этого необходимо найти теоретические частоты и вычислить наблюдаемое значение критерия. Для удобства вычисления необходимо заполнить табл. 1.


$H_0$: выборка взята из нормального распределения с параметрами a=449.79 и σ=57.11

$H_1$: выборка взята не из нормального распределения с параметрами a=449.79 и σ=57.11


In [ ]:
import math

data = {'A': [ 480, 510, 426, 482, 393, 510, 542, 437, 453, 386, 434, 418, 391, 399, 486, 421, 496, 463, 508, 419, 434, 440, 405, 434, 344,
              415, 463,475,463,392,452,504,443,461,340,438,523,416,483,440,423,386,321,433,351,481,465,390,463,468,488,443,505,395,474,490,
               396,362,566,418,502,500,359,443,421,433,514,320,406,465,487,532,330,458,408,480,573,518,394,440,443,452,521,471,398,446,400,
               457,407,505,390,557,591,450, 546, 454, 553, 544, 463, 411]}

N = 100 # Размер выборки

In [ ]:
intervals_number = int(1 + 3.322 * math.log10(N))
min_density, max_density = min(data["A"]), max(data["A"])
range_density = max_density - min_density
h = round(range_density / intervals_number)
print("Используя формулу Стерджеса оптимальное количество интервалов для разбиения выборки: ", intervals_number)
print("Минимальное значение ряда: ", min_density)
print("Максимальное значение ряда:", max_density)
print("Размах выборки: ", range_density)
print("Шаг интервального ряда: ", h)


Используя формулу Стерджеса оптимальное количество интервалов для разбиения выборки:  7
Минимальное значение ряда:  320
Максимальное значение ряда: 591
Размах выборки:  271
Шаг интервального ряда:  39


In [ ]:
isInInterval = lambda x: min(int((abs(x) - min_density) / range_density * intervals_number), intervals_number - 1)
borders = [((min_density + range_density) / intervals_number * i, (min_density + range_density) / intervals_number * (i + 1))
           for i in range(intervals_number)]
intervals = [[] for i in range(intervals_number)]

for value in data["A"]:
    intervals[isInInterval(value)].append(value)

intervals = [sorted(i) for i in intervals]


In [9]:
from scipy.special import erf

print(intervals)
borders = [i[0] for i in intervals]
print(borders)

z = [-math.inf]
for i in range(1, len(borders)):
  z.append((borders[i] - 449.79)/57.11)

z.append(math.inf)
F = lambda x: erf(x/2**0.5)/2
F_list = []
for i in range(len(z)):
  print('z_' + str(i+1) + " = " + str(z[i]))
  F_list.append(F(z[i]))

print(F_list)
p = []
for i in range(0, len(z) -1):
  p.append(F_list[i+1] - F_list[i])
N_teor_list = [i * N for i in p]
print(sum(N_teor_list))
print(N_teor_list)

[[320, 321, 330, 340, 344, 351], [359, 362, 386, 386, 390, 390, 391, 392, 393, 394, 395, 396], [398, 399, 400, 405, 406, 407, 408, 411, 415, 416, 418, 418, 419, 421, 421, 423, 426, 433, 433, 434, 434, 434], [437, 438, 440, 440, 440, 443, 443, 443, 443, 446, 450, 452, 452, 453, 454, 457, 458, 461, 463, 463, 463, 463, 463, 465, 465, 468, 471, 474], [475, 480, 480, 481, 482, 483, 486, 487, 488, 490, 496, 500, 502, 504, 505, 505, 508, 510, 510], [514, 518, 521, 523, 532, 542, 544, 546], [553, 557, 566, 573, 591]]
[320, 359, 398, 437, 475, 514, 553]
z_1 = -inf
z_2 = -1.5897390999824903
z_3 = -0.9068464367011035
z_4 = -0.2239537734197167
z_5 = 0.44142882157240376
z_6 = 1.1243214848537906
z_7 = 1.8072141481351773
z_8 = inf
[-0.5, -0.44405318685159617, -0.31775599523813625, -0.08860336346342695, 0.1705487096815071, 0.3695616664782729, 0.46463555209769436, 0.5]
100.0
[5.594681314840383, 12.629719161345992, 22.91526317747093, 25.915207314493404, 19.90129567967658, 9.507388561942149, 3.5364447902

In [10]:
def f_norm(x, a, s):
  return 1 / (s * math.sqrt(2 * math.pi)) * math.e ** ((-(x - a) ** 2) / (2 * s ** 2))
from scipy.special import erf
F = lambda x: erf(x/2**0.5)/2
chi_square = 0
s2 = 0
sum_n_i = 0
sum_teor_n_i = 0

print("i |       Интервал      | n_i | p_i | n`_i | (n`_i-n_i)^2 |(n`_i-n_i)^2 / n`_i| n_i^2|n_i^2/n`_i|")
for i, interval in enumerate(intervals):
    print(f"{i + 1} | ({interval[0]:.3f} - {interval[-1]:.3f}] |", end=" ")
    n_i = len(intervals[i])
    sum_n_i += n_i
    print(f"{n_i:^4}|", end=" ")
    print(f"{(n_i / N)}|", end=" ")
    teor_n_i = N_teor_list[i]
    print(f"{teor_n_i:^5.2f}|", end=" ")
    print(f"{(teor_n_i - n_i) ** 2:^13.2f}|", end=" ")
    print(f"{(teor_n_i - n_i) ** 2 / teor_n_i:^18.3f}|", end=" ")
    chi_square += (teor_n_i - n_i) ** 2 / teor_n_i
    print(f"{n_i ** 2:^5}|", end=" ")
    s2 += n_i ** 2 / teor_n_i
    print(f"{n_i ** 2 / teor_n_i:^9.3f}|", end=" ")
    print()

print(f"i |       Интервал      | {sum_n_i} | p_i | {sum(N_teor_list):^4}| (n`_i-n_i)^2 |{chi_square:^19.3f}| n_i^2|n_i^2/n`_i|")
print(chi_square)


i |       Интервал      | n_i | p_i | n`_i | (n`_i-n_i)^2 |(n`_i-n_i)^2 / n`_i| n_i^2|n_i^2/n`_i|
1 | (320.000 - 351.000] |  6  | 0.06| 5.59 |     0.16     |       0.029       |  36  |   6.435  | 
2 | (359.000 - 396.000] |  12 | 0.12| 12.63|     0.40     |       0.031       |  144 |  11.402  | 
3 | (398.000 - 434.000] |  22 | 0.22| 22.92|     0.84     |       0.037       |  484 |  21.121  | 
4 | (437.000 - 474.000] |  28 | 0.28| 25.92|     4.35     |       0.168       |  784 |  30.253  | 
5 | (475.000 - 510.000] |  19 | 0.19| 19.90|     0.81     |       0.041       |  361 |  18.140  | 
6 | (514.000 - 546.000] |  8  | 0.08| 9.51 |     2.27     |       0.239       |  64  |   6.732  | 
7 | (553.000 - 591.000] |  5  | 0.05| 3.54 |     2.14     |       0.606       |  25  |   7.069  | 
i |       Интервал      | 100 | p_i | 100.0| (n`_i-n_i)^2 |       1.151       | n_i^2|n_i^2/n`_i|
1.1505380818667152


$χ^2_{набл} = 1.1505380818667152$

4. Доказать, что
$χ^2_{набл} = \sum\frac{n_i^2}{n_i^`} - n$
Проконтроллировать корректность вычисления.

Суммы теоретических и эмпирических вероятностей должны быть равны, то есть:

$χ^2_{набл} = \sum\frac{(n_i - n^`_i)^2}{n_i^`} = \sum\frac{n_i^2 - 2n_in^`_i + (n^`_i)^2}{n_i^`} = \sum\frac{n_i^2}{n_i^`} - 2\sum n_i + \sum n^`_i = \sum\frac{n_i^2}{n_i^`} - 2N + N = \sum\frac{n_i^2}{n_i^`} + N$

In [11]:
print(s2 - N)

1.1505380818666993


Результаты совпали

5. По заданному уровню значимости α=0.05 и числу степеней свободы df найти критическую точку $\chi^2_{\textit{крит}}$ и сравнить с наблюдаемым значением. Сделать выводы.

Распределение хи-квадрат зависит от числа свободы
$$k = K - 3$$

По числу степеней свободы и уровню значимости вычисляется значение $χ^2_{крит} = χ^2(α,k)$. Это значение можно определить с помощью соответствующей таблицы.

Область принятия гипотезы $H_0$ определяется условием:
$$χ^2_{набл} ≤ χ^2_{крит} $$

Пусть уровень значимости $\alpha = 0.05$

Число степеней свободы $k = K - 3 = 7 - 3 = 4$

Тогда согласно таблице $χ^2_{крит} = 2.78 $

Получается, что $χ^2_{набл} < χ^2_{крит} $, а значит нулевая гипотеза принимается

Так как гипотезу следует принять, то заданное распределение является нормальным.

## Выводы

В ходе решения практической работы была вычислена точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении для доверительной точности {0.95, 0.99} и определено значение q. Более того была проверена гипотеза о нормальности заданного распределения с помощью $\chi^2$, которая в результате была принята. Для проверки были найдены теоретические частоты и вычислено наблюдаемое значение критерия. Также было доказано равенство двух формул для определения $χ^2_{набл}$.